### DISCUSSION TITLE
😀how to use the API of AirSim to get the distance from other vehicles added in Unreal 4, road boundaries, and lane center lines to the vehicle?

### DISCUSSION THREAD

**Discussion by Liuzy0908** [microsoft/AirSim#3861] [Support Q&A]
### What's your question?

In UE4, how to use the API of AirSim to get the distance from other cars added in Unreal 4, road boundaries, and lane center lines to the car? anybody could do that to help me?😀

### Include context on what you are trying to achieve

As far as I know, the API of AirSim provided by Microsoft is limited, and I cannot simply implement the functions mentioned above with python. Now I am using Unreal 4 engine for scene development to make it easy to obtain the data needed in reinforcement learning, but the method of obtaining the above data is urgently needed. The idea that I can think of is to use the blueprint function in Unreal 4, but it is too complicated and cumbersome.

**Comment by jonyMarino** [comment]
Hi and welcome @Liuzy0908!
You can get the object position with simGetObjectPose() and then get the car position with getVehiclePose() and calculate the absolute distance between the two of them.
